libvast/src/system/configuration.cpp

```diff
@@ -444,12 +444,24 @@ caf::error configuration::parse(int argc, char** argv) {
         .add<std::vector<std::string>>("?vast", "schema-dirs", "")
         .add<std::vector<std::string>>("?vast", "plugin-dirs", "")
         .add<std::vector<std::string>>("?vast", "plugins", "");
-  auto [ec, it] = plugin_opts.parse(content, plugin_args);
-  if (ec != caf::pec::success) {
-    VAST_ASSERT(it != plugin_args.end());
-    return caf::make_error(ec, fmt::format("failed to parse option '{}'", *it));
+
+  // edge case for "plugins" option: no argument specified (e.g. "--plugins=")
+  // this case should then be resolved to "load no plugins" (empty list)
```

Comment:
Is there a way that we can fix this generally instead of adding a one-off for a single option?

Reply:
1. Generally for all plugin-only options: Yes, I can remove the `starts_with("--plugins=")` check.
2. Generally for all options: Well, `CAF` does the parsing for us in `config_option_set.cpp`, I probably would have to implement workarounds like this one...

Reply:
... or maybe a helper function/method.

Reply:
... or I could try to iterate through all command arguments instead of only `plugin_args`.

Reply:
I think either works, but ultimately the option handling should be consistent for options of the same type. In this particular case, I'd expect all options that contain a list of strings to behave the same. If I can specify them as a comma-separated list then it'd only be natural for the empty string to be parsed as an empty list.

Reply:
Wow, implementing this took a bit of time.. the problem is we have multiple `parse` ings in the `vast` application setup and also the setup workflow steps can't be just moved around (plugin arguments need to be parsed before loading them and then parsing arguments again).

But I think I managed to implement an approach.